library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -61,6 +62,49 @@ class Client() {
         fun register(codec: ContentCodec<*>) {
             codecRegistry.register(codec = codec)
         }
+
+        /**
+         * Use the {@param api} to fetch any stored keys belonging to {@param address}.
+         *
+         * The user will need to be prompted to sign to decrypt each bundle.
+         */
+        suspend fun authCheck(api: ApiClient, address: String): List<EncryptedPrivateKeyBundle> {
+            val topic: Topic = Topic.userPrivateStoreKeyBundle(toChecksumAddress(address))
+            val res = api.queryTopic(topic)
+            return res.envelopesList.mapNotNull {
+                try {
+                    EncryptedPrivateKeyBundle.parseFrom(it.message)
+                } catch (e: Exception) {
+                    print("discarding malformed private key bundle: ${e.message}")
+                    null
+                }
+            }
+        }
+
+        /**
+         * Use the {@param api} to save the {@param encryptedKeys} for {@param address}.
+         *
+         * The {@param keys} are used to authorize the publish request.
+         */
+        suspend fun authSave(
+            api: ApiClient,
+            keys: PrivateKeyBundle,
+            encryptedKeys: EncryptedPrivateKeyBundle,
+        ) {
+            val authorizedIdentity = AuthorizedIdentity(keys.v1)
+            authorizedIdentity.address = keys.v1.walletAddress
+            val authToken = authorizedIdentity.createAuthToken()
+            api.setAuthToken(authToken)
+            api.publish(
+                envelopes = listOf(
+                    EnvelopeBuilder.buildFromTopic(
+                        topic = Topic.userPrivateStoreKeyBundle(keys.v1.walletAddress),
+                        timestamp = Date(),
+                        message = encryptedKeys.toByteArray()
+                    )
+                )
+            )
```

Comment:
Feels like there could be some optimization here instead of calling `keys.v1` 3 different times. Maybe authSave could take v1Keys instead of a PrivateKeyBundle.